docs/src/docs/asciidoc/maven-plugin-quickstart.adoc

```diff
@@ -0,0 +1,459 @@
++
+The executable's name is derived from the artifact ID, but you can specify any custom name in `native-maven-plugin` within a
+`<configuration>` node:
++
+[source,xml]
+----
+<configuration>
+    <imageName>fortuneteller</imageName>
+</configuration>
+----
+. Run the demo directly or with the Maven profile:
++
+[source,shell]
+----
+./target/fortune
+----
++
+[source,shell]
+----
+mvn -Pnative exec:exec@native
+----
+
+To see the benefits of running your application as a native executable,
+`time` how long it takes and compare the results with running on the
+JVM.
+
+== Add JUnit Testing
+
+The Maven plugin for GraalVM Native Image can run
+https://junit.org/junit5/docs/current/user-guide/[JUnit Platform] tests on a native executable. This means that tests will be compiled and executed as native code.
+
+This plugin requires JUnit Platform 1.8 or higher and Maven Surefire 2.22.0 or higher to run tests on a native executable.
+
+. Enable extensions in the plugin's configuration,
+`<extensions>true</extensions>`:
++
+[source,xml]
+----
+<plugin>
+    <groupId>org.graalvm.buildtools</groupId>
+    <artifactId>native-maven-plugin</artifactId>
+    <version>${native.maven.plugin.version}</version>
+    <extensions>true</extensions>
+----
+. Add an explicit dependency on the `junit-platform-launcher` artifact
+to the dependencies section of your native profile configuration as in
+the following example:
++
+[source,xml]
+----
+<dependencies>
+    <dependency>
+        <groupId>org.junit.platform</groupId>
+        <artifactId>junit-platform-launcher</artifactId>
+        <version>1.8.2</version>
+        <scope>test</scope>
+    </dependency>
+</dependencies>
+----
+. Create the following test in the
+`src/test/java/demo/FortuneTest.java` file:
++
+.src/test/java/demo/FortuneTest.java
+[source,java]
+----
+package demo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class FortuneTest {
+    @Test
+    @DisplayName("Returns a fortune")
+    void testItWorks() throws JsonProcessingException {
+        Fortune fortune = new Fortune();
+        assertTrue(fortune.randomFortune().length()>0);
+    }
+}
+----
++
+. Run native tests:
++
+[source,shell]
+----
+mvn -Pnative test
+----
++
+Run `-Pnative` profile will then build and run native tests.
+
+=== Summary
+
+The Maven plugin for GraalVM Native Image adds support for building and testing native executables using https://maven.apache.org/[Apache Maven™]. The plugin has many features, described in the <<maven-plugin.adoc#,plugin reference documentation>>.
+
+Note that if your application does not call any classes dynamically at run time, the execution with the agent is needless. Your workflow, in that case, is just:
+
+[source,shell]
+----
+mvn clean compile
+mvn -Pnative package
```

Comment:
Can't you just do `mvn clean package -Pnative`? `package` will invoke `compile`, right?

Reply:
Right, one command is sufficient: `mvn clean -Pnative package`.